src/outputs/parthenon_hdf5.cpp

```diff
@@ -868,9 +926,16 @@ void PHDF5Output::WriteOutputFileImpl(Mesh *pm, ParameterInput *pin, SimTime *tm
 
       if (!is_allocated) {
         if (vinfo.is_sparse) {
-          hsize_t N = varSize * vlen;
-          memset(tmpData.data() + index, 0, N * sizeof(OutT));
-          index += N;
+          hsize_t varSize{};
+          if (vinfo.where == MetadataFlag(Metadata::Cell)) {
+            varSize = vinfo.nx6 * vinfo.nx5 * vinfo.nx4 * (out_kb.e - out_kb.s + 1) *
+                      (out_jb.e - out_jb.s + 1) * (out_ib.e - out_ib.s + 1);
```

Comment:
Why not just use the sizes contained in `vinfo` and get rid of the `if`?

Reply:
Using the sizes in `vinfo` was actually a bug that broke the `sparse_advection` restart test that I just fixed earlier here -- the `vinfo` dimensions contain ghost zones but to remove those we need e.g. `out_kb.e - out_kb.s + 1`. Whereas for non-`Metadata::Cell` variables, we don't in general know that e.g. `out_kb.e - out_kb.s + 1` is the appropriate subset of `nx3`.